src/java.base/share/classes/javax/crypto/Cipher.java

```diff
@@ -96,49 +96,29 @@
  * provide authenticity assurances for both confidential data and
  * Additional Associated Data (AAD) that is not encrypted.  (Please see
  * <a href="http://www.ietf.org/rfc/rfc5116.txt"> RFC 5116 </a> for more
- * information on AEAD and AAD algorithms such as GCM/CCM.) Both
+ * information on AEAD and AAD algorithms.) Both
  * confidential and AAD data can be used when calculating the
  * authentication tag (similar to a {@link Mac}).  This tag is appended
  * to the ciphertext during encryption, and is verified on decryption.
  * <p>
- * AEAD modes such as GCM/CCM perform all AAD authenticity calculations
+ * AEAD modes perform all AAD authenticity calculations
  * before starting the ciphertext authenticity calculations.  To avoid
  * implementations having to internally buffer ciphertext, all AAD data
- * must be supplied to GCM/CCM implementations (via the {@code updateAAD}
+ * must be supplied to their implementations (via the {@code updateAAD}
  * methods) <b>before</b> the ciphertext is processed (via
  * the {@code update} and {@code doFinal} methods).
  * <p>
- * Note that GCM mode has a uniqueness requirement on IVs used in
- * encryption with a given key. When IVs are repeated for GCM
- * encryption, such usages are subject to forgery attacks. Thus, after
- * each encryption operation using GCM mode, callers should re-initialize
- * the {@code Cipher} objects with GCM parameters which have a different IV
- * value.
- * <pre>
- *     GCMParameterSpec s = ...;
- *     cipher.init(..., s);
- *
- *     // If the GCM parameters were generated by the provider, it can
- *     // be retrieved by:
- *     // cipher.getParameters().getParameterSpec(GCMParameterSpec.class);
- *
- *     cipher.updateAAD(...);  // AAD
- *     cipher.update(...);     // Multi-part update
- *     cipher.doFinal(...);    // conclusion of operation
+ * When {@code doFinal} methods completes the operation, the {@code Cipher} object will attempt
+ * to reset the state to the most recent call to {@code init}, allowing for additional
+ * operations. A successful reset depends on the mode ({@code ENCRYPT_MODE} or
+ * {@code DECRYPT_MODE}) and the algorithm. AEAD algorithms may not reset to prevent
+ * forgery attacks due to Key and IV uniqueness requirements.
+ * An {@link IllegalStateException} will be thrown when calling {@code update}
+ * or {@code doFinal} methods when a reset did not occur.
```

Comment:
It might be helpful to describe what to do instead:

When reset is not possible, you don't need to dispose the cipher object. Instead you can call init again with a different key/nonce.